pulsar/internal/connection.go

```diff
@@ -550,16 +550,19 @@ func (c *connection) handleCloseConsumer(closeConsumer *pb.CommandCloseConsumer)
 	consumerID := closeConsumer.GetConsumerId()
 	if consumer, ok := c.consumerHandler(consumerID); ok {
 		consumer.ConnectionClosed()
+		delete(c.listeners, consumerID)
```

Comment:
Should we use this function DeleteConsumeHandler to delete the handler? We get the handler using the consumerHandlersLock so to be consistent should we delete with it too?

Reply:
Seems like this is a different map so it ok.